lib/mime.c

```diff
@@ -279,29 +280,66 @@ static void mimesetstate(struct mime_state *state,
 
 
 /* Escape header string into allocated memory. */
-static char *escape_string(const char *src)
+static CURLcode escape_mime_string(struct dynbuf *db, const char *src)
 {
-  size_t bytecount = 0;
-  size_t i;
-  char *dst;
+  CURLcode result = CURLE_OK;
 
-  for(i = 0; src[i]; i++)
-    if(src[i] == '"' || src[i] == '\\')
-      bytecount++;
+  for(; !result && *src; result = Curl_dyn_addn(db, src++, 1))
+    if(*src == '\\' || *src == '"') {
+      result = Curl_dyn_add(db, "\\");
+      if(result)
+        break;
+    }
+  return result;
+}
 
-  bytecount += i;
-  dst = malloc(bytecount + 1);
-  if(!dst)
-    return NULL;
+static CURLcode escape_form_string(struct Curl_easy *easy,
+                                   struct dynbuf *db, const char *src)
+{
+  CURLcode result = CURLE_OK;
+
+  (void) easy;
 
-  for(i = 0; *src; src++) {
-    if(*src == '"' || *src == '\\')
-      dst[i++] = '\\';
-    dst[i++] = *src;
+  for(; !result && *src; src++) {
+    if(*src != '"' && *src != '\r' && *src != '\n')
```

Comment:
Maybe a comment here with an explanation/reference to why it only encodes these three particular bytes?

Reply:
Will do.

Reply:
Done.